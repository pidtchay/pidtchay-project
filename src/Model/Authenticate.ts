/**
 * Login data model.
 * @property [string] nickName - user nickName
 * @property [string] email - user email
 * @property [string] password - user password
 */
export interface ILoginData {
  nickName: string;
  email: string;
  password: string;
}

/**
 * @property [boolean] authenticated
 * @property [string] session
 * @property [string] userName
 */
export interface ISystemState {
  authenticated: boolean;
  session: string;
  userName: string;
}
