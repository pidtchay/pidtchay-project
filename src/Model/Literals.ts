/**
 * Data model for internationalization.
 */
export interface ILiterals {
    [key: string]: string | ILiterals;
}
