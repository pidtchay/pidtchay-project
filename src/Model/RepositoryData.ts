/**
 * Data model for query result on github api.
 *
 * @property {string} [avatarImg] Avatag image.
 * @property {string} [description] Repository description.
 * @property {string} [forks] Repository forks.
 * @property {string} [homepage] Repository homepage.
 * @property {number} [id] Id.
 * @property {string} [language] Used programming language.
 * @property {string} [name] Repository name.
 * @property {string} [owner] Repository owner.
 * @property {string} [type] Type.
 * @property {string} [watchers] Repository watchers.
 */
export interface IRepositoryData {
    avatarImg: string;
    description: string;
    forks: number;
    homepage: string;
    id: number;
    language: string;
    name: string;
    owner: string;
    type: string;
    watchers: number;
}
