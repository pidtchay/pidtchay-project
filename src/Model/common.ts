/**
 * Data model for notification window.
 * @property {string} [description] Description of the event that triggered the notification window.
 * @property {string} [title] Notification window title.
 * @property type Type of notification displayed.
 */
export interface INotificationProps {
  description: string;
  title: string;
  type: 'success' | 'info' | 'warning' | 'error';
}
