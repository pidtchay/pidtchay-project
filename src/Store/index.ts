import { combineReducers } from 'redux';
import { literalsReducer } from './literals/reducers';
import { reducer } from './markdown_notes/reducer';
import { repositoryReducer } from './repository/reducers';
import { systemReducer } from './system/reducers';

export const rootReducer = combineReducers({
    system: systemReducer,
    repository: repositoryReducer,
    literals: literalsReducer,
    markdownNotes: reducer
});

export type RootState = ReturnType<typeof rootReducer>;
