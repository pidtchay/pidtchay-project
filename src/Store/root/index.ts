import { combineReducers } from 'redux';
import { languagesReducer } from 'Store/languages/reducers';
import { listRepositoriesSlice } from 'Store/repositories/slice';
// import { listRepositoriesReducer } from 'Store/repositories/reducers';

export const newRootReducer = combineReducers({
    githubRepos: listRepositoriesSlice.reducer,
    languages: languagesReducer
    // githubRepos: listRepositoriesReducer
});
export type NewRootState = ReturnType<typeof newRootReducer>;
