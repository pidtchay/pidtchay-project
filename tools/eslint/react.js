module.exports = {
    'react/no-did-update-set-state': 0,
    'react-hooks/exhaustive-deps': 0,
    'react/state-in-constructor': [2, 'never'],
    'react/destructuring-assignment': 0,
    'react/no-danger': 2,
    'react/prefer-stateless-function': 1,
    'react/no-array-index-key': 1,
    'react/jsx-props-no-spreading': 0,
    'react/static-property-placement': ['warn', 'property assignment', { // Указывает, где должны быть размещены статические свойства компонента React
      'childContextTypes': 'static getter',
      'contextTypes': 'static public field',
      'contextType': 'static public field',
      'displayName': 'static public field',
    }],
    'react/require-default-props': 1,
    'react/sort-comp': [1, { order: ['static-methods', 'lifecycle', 'everything-else', 'render']}],

}