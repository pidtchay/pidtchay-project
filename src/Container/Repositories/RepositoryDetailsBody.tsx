import {
    Divider,
    Row,
    Col,
    Avatar,
    Descriptions,
    Tag,
    Button,
    PageHeader
} from 'antd';
import * as React from 'react';
import { IRepositoryData } from 'Model/RepositoryData';
import style from 'Style/RepositoryDetails.less';
import { useLiteralValue } from 'Utils/hooks';

interface IProps {
    repository: IRepositoryData;
    onBackClick: () => void;
}

export const RepositoryDetails: React.FC<IProps> = ({
    repository,
    onBackClick: handleBackClick
}) => {
    const { getValue: getLiteralValue } = useLiteralValue();
    return (
        <div className={style.details_layout}>
            <PageHeader
                onBack={handleBackClick}
                title={getLiteralValue('Pages.GithubRepositories.back') || ''}
            />
            <Divider orientation="left">
                {getLiteralValue(
                    'Pages.GithubRepositories.Details.Divider.owner'
                )}
            </Divider>
            <Row gutter={[16, 16]}>
                <Col span={4}>
                    <div className={style.avatar_panel}>
                        <Avatar
                            shape="square"
                            size={100}
                            src={repository.avatarImg}
                        />
                    </div>
                </Col>
                <Col span={10} offset={2}>
                    <Descriptions
                        title={getLiteralValue(
                            'Pages.GithubRepositories.Details.info'
                        )}
                    >
                        <Descriptions.Item
                            label={getLiteralValue(
                                'Pages.GithubRepositories.Details.owner'
                            )}
                        >
                            {repository.owner}
                        </Descriptions.Item>
                        <Descriptions.Item
                            label={getLiteralValue(
                                'Pages.GithubRepositories.Details.home_page'
                            )}
                        >
                            <Button type="link">{repository.homepage}</Button>
                        </Descriptions.Item>
                        <Descriptions.Item
                            label={getLiteralValue(
                                'Pages.GithubRepositories.Details.forks'
                            )}
                        >
                            {repository.forks}
                        </Descriptions.Item>
                        <Descriptions.Item
                            label={getLiteralValue(
                                'Pages.GithubRepositories.Details.watchers'
                            )}
                        >
                            {repository.watchers}
                        </Descriptions.Item>
                    </Descriptions>
                </Col>
            </Row>
            <Divider orientation="left">
                {getLiteralValue('Pages.GithubRepositories.Divider.details')}
            </Divider>
            <Row gutter={[16, 16]}>
                <Col>
                    <Descriptions
                        title={getLiteralValue(
                            'Pages.GithubRepositories.Details.description'
                        )}
                    >
                        <Descriptions.Item
                            label={getLiteralValue(
                                'Pages.GithubRepositories.Details.name'
                            )}
                        >
                            {repository.name}
                        </Descriptions.Item>
                        <Descriptions.Item
                            label={getLiteralValue(
                                'Pages.GithubRepositories.Details.description'
                            )}
                        >
                            {repository.description}
                        </Descriptions.Item>
                        <Descriptions.Item
                            label={getLiteralValue(
                                'Pages.GithubRepositories.Details.language'
                            )}
                        >
                            <Tag color="geekblue">{repository.language}</Tag>
                        </Descriptions.Item>
                    </Descriptions>
                </Col>
            </Row>
        </div>
    );
};
