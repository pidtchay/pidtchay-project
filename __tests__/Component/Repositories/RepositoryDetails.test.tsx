import * as React from 'react';
import { Provider } from 'react-redux';
import * as renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { RepositoryDetails } from 'Container/Repositories/Details/RepositoryDetailsBody';
import { IRepositoryData } from 'Model/RepositoryData';
import { newRootReducer } from 'Store/root';

const store = createStore(newRootReducer);

describe('render success', () => {
    it('should match the snapshot', async () => {
        const handleBackClick = jest.fn();
        const repository: IRepositoryData = {
            avatarImg: 'http://test.com/avatar.jpeg',
            description: '',
            forks: 1,
            homepage: 'http://test.com/home',
            id: 1,
            language: 'javascript',
            name: 'ReactJS Todo list',
            owner: 'Monster corp',
            type: '',
            watchers: 5
        };
        const component = renderer.create(
            <Provider store={store}>
                <RepositoryDetails onBackClick={handleBackClick} repository={repository} />
            </Provider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
