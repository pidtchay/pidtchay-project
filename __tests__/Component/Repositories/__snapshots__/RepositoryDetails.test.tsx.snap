// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render success should match the snapshot 1`] = `
<div>
  <div
    className="ant-page-header ant-page-header-ghost"
  />
  <div
    className="ant-divider ant-divider-horizontal"
    role="separator"
  />
  <div
    className="ant-row"
    style={
      Object {
        "marginBottom": 8,
        "marginLeft": -8,
        "marginRight": -8,
        "marginTop": -8,
      }
    }
  >
    <div
      className="ant-col ant-col-4"
      style={
        Object {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
    >
      <div>
        <span
          className="ant-avatar ant-avatar-square ant-avatar-image"
          style={
            Object {
              "fontSize": 18,
              "height": 100,
              "lineHeight": "100px",
              "width": 100,
            }
          }
        >
          <img
            onError={[Function]}
            src="http://test.com/avatar.jpeg"
          />
        </span>
      </div>
    </div>
    <div
      className="ant-col ant-col-10 ant-col-offset-2"
      style={
        Object {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
    >
      <div
        className="ant-descriptions"
      >
        <div
          className="ant-descriptions-view"
        >
          <table>
            <tbody>
              <tr
                className="ant-descriptions-row"
              >
                <td
                  className="ant-descriptions-item"
                  colSpan={1}
                >
                  <span
                    className="ant-descriptions-item-content"
                  >
                    Monster corp
                  </span>
                </td>
                <td
                  className="ant-descriptions-item"
                  colSpan={1}
                >
                  <span
                    className="ant-descriptions-item-content"
                  >
                    <button
                      className="ant-btn ant-btn-link"
                      onClick={[Function]}
                      type="button"
                    >
                      <span>
                        http://test.com/home
                      </span>
                    </button>
                  </span>
                </td>
                <td
                  className="ant-descriptions-item"
                  colSpan={1}
                >
                  <span
                    className="ant-descriptions-item-content"
                  >
                    1
                  </span>
                </td>
              </tr>
              <tr
                className="ant-descriptions-row"
              >
                <td
                  className="ant-descriptions-item"
                  colSpan={3}
                >
                  <span
                    className="ant-descriptions-item-content"
                  >
                    5
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-divider ant-divider-horizontal"
    role="separator"
  />
  <div
    className="ant-row"
    style={
      Object {
        "marginBottom": 8,
        "marginLeft": -8,
        "marginRight": -8,
        "marginTop": -8,
      }
    }
  >
    <div
      className="ant-col"
      style={
        Object {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
    >
      <div
        className="ant-descriptions"
      >
        <div
          className="ant-descriptions-view"
        >
          <table>
            <tbody>
              <tr
                className="ant-descriptions-row"
              >
                <td
                  className="ant-descriptions-item"
                  colSpan={1}
                >
                  <span
                    className="ant-descriptions-item-content"
                  >
                    ReactJS Todo list
                  </span>
                </td>
                <td
                  className="ant-descriptions-item"
                  colSpan={1}
                >
                  
                </td>
                <td
                  className="ant-descriptions-item"
                  colSpan={1}
                >
                  <span
                    className="ant-descriptions-item-content"
                  >
                    <span
                      className="ant-tag ant-tag-geekblue"
                      style={
                        Object {
                          "backgroundColor": undefined,
                        }
                      }
                    >
                      javascript
                    </span>
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
